modules/core/src/main/scala/gem/syntax/Parser.scala

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.syntax
+
+import atto._, Atto._
+
+final class ParserOps[A](val self: Parser[A]) extends AnyVal {
+
+  /** Parse entire input into an Option. */
+  def parseExact(s: String): Option[A] =
+    (self <~ endOfInput).parseOnly(s).option
```

Comment:
Very clever 👍. Does it backtrack if it fails? What this will return? `(parseExact("foo") | parseExact("bar")).parseExact("bar")`